staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -105,13 +109,14 @@ func NewNamedReflector(name string, lw ListerWatcher, expectedType interface{},
 	r := &Reflector{
 		name: name,
 		// we need this to be unique per process (some names are still the same) but obvious who it belongs to
-		metrics:       newReflectorMetrics(makeValidPrometheusMetricLabel(fmt.Sprintf("reflector_"+name+"_%d", reflectorSuffix))),
-		listerWatcher: lw,
-		store:         store,
-		expectedType:  reflect.TypeOf(expectedType),
-		period:        time.Second,
-		resyncPeriod:  resyncPeriod,
-		clock:         &clock.RealClock{},
+		metrics:            newReflectorMetrics(makeValidPrometheusMetricLabel(fmt.Sprintf("reflector_"+name+"_%d", reflectorSuffix))),
+		listerWatcher:      lw,
+		store:              store,
+		expectedType:       reflect.TypeOf(expectedType),
+		period:             time.Second,
+		resyncPeriod:       resyncPeriod,
+		clock:              &clock.RealClock{},
+		consistentErrorsCh: make(chan error),
```

Comment:
intentionally unbuffered?

Reply:
yes, I think we don't need a buffered channel.

Reply:
> yes, I think we don't need a buffered channel.

Based on my view, I don't see a need for readers to block writers.  Is there a need for that behavior that I'm not seeing?  If we don't need it, I'd like a small buffer. Say 20.

Reply:
+1

Reply:
Do we ever close consistentErrorsCh? I noticed we check for it being closed on the read side but I can't find where we close the channel.

Reply:
no, we don't but I think it's okay to leave it open. It will be created on every call to `informer.Run` which usually happens only once at startup and `GCed` when the old informer is no longer needed.

The receiver checks for the closed channel because `nil` value has a special meaning (reset counter).